pkg/reconciler/route/resources/http_routes_builder.go

```diff
@@ -0,0 +1,434 @@
+package resources
```

Comment:
nit: Is it possible to do these in place then move the file in a later PR? That minimizes the diff so we can be more confident that nothing major changed

a big file is okay in go as well so a cleanup PR isn't strictly necessary